pkg/plugin/system/system.go

```diff
@@ -45,6 +46,11 @@ type pluginSystem struct {
 	clients       []ClientProvider
 	plugins       *PluginList
 	promptRunner  ioutils.PromptRunner
+	callbackMap   map[string]CustomCommandCallbackFn
+}
+
+type CustomCommandCallbackFn interface {
```

Comment:
Why an interface instead of just a plain type (alias? not sure what it would be called). If it's an interface that wraps a callback I don't think the name itself should be called a "callback function"?

Reply:
I wanted the function `CustomCommandCallback` to be available when casted to this type. There definitely may be a better way to set this up. If you know how to change it so it doesnt use the interface then I am happy to change it. I just wasnt sure how to do that myself in this case